### DISCUSSION TITLE
When using search filter on table editor, strings ending with whitespace cannot be searched for with `=` operator.

### DISCUSSION THREAD

**Discussion by dxrsz** [supabase/supabase#16934] [Contribute to Supabase]
# Bug report

## Describe the bug

When using search filter on table editor, strings ending with whitespace cannot be searched for with `=`.

## To Reproduce

* Create a table with a "text" column (e.g., "name")
* Insert a row with `name` assigned to `"test "` <- note the terminating space
* Observe the row appearing in the table editor
* Add a filter for name = `test `
* Observe no results returned

## Expected behavior

Row with terminating spaces should be returned if searched for.

## Screenshots
n/a

## System information

- OS: win 10
- Browser (if applies) chrome
- Version of supabase-js: [e.g. 6.0.2] n/a
- Version of Node.js: [e.g. 10.10.0] n/a

## Additional context

Might be related to the fact that if you right-click / copy a cell with a terminating space, it leave out the space.

**Comment by oxenprogrammer** [comment]
@evanandrewrose  Cc: @joshenlim  What do you think about `trimming string fields` to remove trailing/leading spaces whenever adding/updating new rows, that way we have a more standard approach since I see we are currently trimming the filter input [here](https://github.com/supabase/supabase/blob/1271356aa6df39c76a5eaf81281427c4069e03b1/studio/components/grid/query/Query.utils.ts#L206) on `filterLiteral` function before performing the filtering.

**Comment by dxrsz** [comment]
> remove trailing/leading spaces whenever adding/updating new rows

It sounds like that would prevent users from writing strings that end with spaces, which is needed for my use case. Plus, I would assume you'd still be able to get values ending in whitespace into the database through SQL inserts (how my data actually gets written).

IMO, the change should be on the filtering side to support matching whitespace characters. The output from the UI should match the output from `select * from test where name = 'test '`.

**Comment by oxenprogrammer** [comment]
That makes sense, thanks for clarifying. I will explore the option of leaving filter input untrimmed to see if there are any unexpected side effects.
